src/overlays/actors/ovl_En_Yukabyun/z_en_yukabyun.c

```diff
@@ -27,19 +30,123 @@ const ActorInit En_Yukabyun_InitVars = {
+    { 28, 8, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 16, ICHAIN_STOP),
+};
+
+s32 D_80B43F64[] = {
+    0x06000AF0, 0x06000000, 0x00000000
+};
+
+extern UNK_TYPE D_06000A60;
+extern UNK_TYPE D_06000040;
+extern Gfx D_06000970[];
+
+void EnYukabyun_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnYukabyun* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 0.4f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.params++;
+    this->unk_152 = 0;
+    this->unk_150 = (u8)(this->actor.params) * 0xA + 0x14;
+    this->actionfunc = func_80B43A94;
+}
+
+void EnYukabyun_Destroy(Actor *thisx, GlobalContext *globalCtx) {
+    EnYukabyun* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80B43A94(EnYukabyun* this, GlobalContext* globalCtx) {
+    if (this->unk_150 != 0) {
+        this->unk_150--;
+    }
+    if (this->unk_150 == 0) {
+        this->actor.flags |= 0x1005;
+        this->actionfunc = func_80B43AD4;
+    }
+}
+
+void func_80B43AD4(EnYukabyun* this, GlobalContext* globalCtx) {
+    this->unk_150 += 0xA0;
+    this->actor.shape.rot.y += this->unk_150;
+    if (this->unk_150 >= 0x2000) {
+        this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+        this->actor.speedXZ = 10.0f;
+        this->actionfunc = func_80B43B6C;
+    }
+    Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y + 30.0f, 1.0f);
+    func_8002F974(&this->actor, NA_SE_EN_YUKABYUN_FLY - SFX_FLAG);
+}
+
+void func_80B43B6C(EnYukabyun* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += this->unk_150;
+    if (this->actor.xzDistFromLink > 5000.0f) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    func_8002F974(&this->actor, NA_SE_EN_YUKABYUN_FLY - SFX_FLAG);
+}
+
+void func_80B43BCC(EnYukabyun* this, GlobalContext* globalCtx) {
+    func_800297A4(globalCtx, &this->actor.posRot.pos, 8.0f, 0, 0x514, 0x12C, 0xF, 0x5F, 0xA, &D_06000A60);
+    Actor_Kill(&this->actor);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/func_80B43A94.s")
+void EnYukabyun_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnYukabyun* this = THIS;
+    s32 pad;
+    
+    if (((this->collider.base.atFlags & 2) || (this->collider.base.acFlags & 2) ||
+        ((this->collider.base.maskA & 2) && !(this->collider.base.oc->id == ACTOR_EN_YUKABYUN))) ||
+        ((this->actionfunc == func_80B43B6C) && (this->actor.bgCheckFlags & 8)))
+        {
+            this->collider.base.atFlags &= 0xFFFD;
+            this->collider.base.acFlags &= 0xFFFD;
+            this->collider.base.maskA &= 0xFFFD;
+            this->actor.flags &= -6;
```

Comment:
for these youll want to turn them into their "not" version
you can just plug it into your calculator and select not. 0xFFFD is ~0x2
mips2c incorrectly made that last one a negative number, which is really 0xFFFA or ~0x5